cpp/src/parquet/reader_test.cc

```diff
@@ -547,12 +547,12 @@ Column 1: b (INT32)
 --- Rows: 3 ---
 Column 0
   Values: 18  Statistics Not Set
-  Compression: SNAPPY, Encodings: RLE PLAIN_DICTIONARY
+  Compression: SNAPPY, Encodings: PLAIN_DICTIONARY(DICT_PAGE) PLAIN_DICTIONARY
```

Comment:
This is weird. The same data was RLE-encoded before this PR and suddenly it is plain-dict encoded?

Reply:
Perhaps RLE is how dictionary indices are encoded... but then why is this information not displayed anymore?

Reply:
(but then it would have been RLE_DICTIONARY?)

Reply:
The RLE encoding was used for def/rep levels. After this change, it only prints encoding of data. This is in sync with what parquet-mr does: https://github.com/apache/parquet-mr/blob/c00479d538353348726cf78da835078024161e55/parquet-cli/src/main/java/org/apache/parquet/cli/commands/ParquetMetadataCommand.java#L163

Reply:
So the main idea is that we should always prefer encoding_stats to encoding if encoding_stats exists when printing encoding metadata.

Reply:
I see, thanks.